core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -444,100 +444,90 @@ struct SubviewExtents {
   size_t m_length[InternalRangeRank];
   unsigned m_index[InternalRangeRank];
 
-  template <size_t... DimArgs>
-  KOKKOS_FORCEINLINE_FUNCTION bool set(unsigned, unsigned,
-                                       const ViewDimension<DimArgs...>&) {
+  template <int DRank, int RRank, size_t... DimArgs>
```

Comment:
can we please keep the name `DomainRank` and `RangeRank`?

Reply:
No, these are already template parameter names of this class.

Reply:
what do you mean "already"?

Reply:
```
$ git grep -n "DomainRank"
core/src/impl/Kokkos_ViewMapping.hpp:435:template <unsigned DomainRank, unsigned RangeRank>
core/src/impl/Kokkos_ViewMapping.hpp:443:  size_t m_begin[DomainRank];
core/src/impl/Kokkos_ViewMapping.hpp:660:    static_assert(DomainRank == sizeof...(DimArgs), "");
core/src/impl/Kokkos_ViewMapping.hpp:661:    static_assert(DomainRank == sizeof...(Args), "");
core/src/impl/Kokkos_ViewMapping.hpp:687:    return unsigned(i) < DomainRank ? m_begin[i] : 0;
```
The template parameters for `SubviewExtents` are called `DomainRank` and `RangeRank`. Thus, we can't use those names for the `set` member functions but would need to find a different name.

Reply:
ah sorry i thought you meant "already" as in they were already temp params of the method